### PR TITLE
Enhance 275 GTSAM localizer relative estimate

### LINKED ISSUE #275: Modify combined state formulation for wave_gtsam.
In order to decouple scan matches and the constant velocity model, the states need to change from

Pose_ODOM_BL
Vel_ODOM_BL
GPS_BIAS 

to

T_ODOM_BL
T_MAP_ODOM
Vel_ODOM_BL

and an additional 
T_ECEF_MAP

The GPS, Lidar, and propagation residuals need to be reformulated for this new state, and the initialization inside the localizer needs to be updated to reflect this new state. In effect, this change is implementing REP-105 for frame conventions. It should move any jitters in the output due to scan matches to T_MAP_ODOM to ensure that the motion model smoothness is only enforced on the sequence of T_ODOM_BL.

### PR DESCRIPTION
**Pull request by navganti**
Resolves #275.

The issue description may be misleading - this started as a reformulation of the GTSAM state for the localizer, however after more discussion all that was required is a modification of the PoseVel state and the motion factor.

Changes:
- Templated motion factor to work on PoseVel or PoseVelBias.
- Cleaned up some compilation issues with PoseVel.
- Added PoseVel tests.

#### Pre-Merge Checklist:
- [x] Code is documented in doxygen format
- [x] Code has automated tests
- [x] Zero compiler warnings
- [x] Formatted with `clang-format`

### CODE REVIEW

File: wave_gtsam/include/wave/gtsam/pose_vel_bias.hpp [hunk 1]
```diff
@@ -151,8 +151,8 @@ struct traits<wave::PoseVelBias> {
             J3.resize(3, 3);
         }
         retval.pose = traits<Pose3>::Expmap(v.block<6, 1>(0, 0), J1);
-        retval.vel = traits<VelType>::Logmap(v.block<6, 1>(6, 0), J2);
-        retval.bias = traits<BiasType>::Logmap(v.block<3, 1>(12, 0), J3);
+        retval.vel = traits<VelType>::Expmap(v.block<6, 1>(6, 0), J2);
+        retval.bias = traits<BiasType>::Expmap(v.block<3, 1>(12, 0), J3);
```
Comment (Jebediah) [inline wave_gtsam/include/wave/gtsam/pose_vel_bias.hpp]: Wow. Whoops

**Comment by Jebediah** [comment]
The code duplication in motion factor is kind of annoying, but I don't know how to fix it so whatever.

**Comment by navganti** [comment]
@Jebediah yeah - I tried to get around that by templating but I had to define the Jacobian dimensions so I figured this was fastest.

**Comment by navganti** [comment]
Thanks guys. I'm going to merge this now as I want the submodule to be updated for autonomoose.

**Pull request merged**